bitcoin/src/pow.rs

```diff
@@ -138,6 +138,19 @@ impl Target {
     // https://github.com/bitcoin/bitcoin/blob/8105bce5b384c72cf08b25b7c5343622754e7337/src/kernel/chainparams.cpp#L348
     pub const MAX_ATTAINABLE_SIGNET: Self = Target(U256(0x0377_ae00 << 80, 0));
 
+    /// Returns the maximum attainable target for `network`.
+    ///
+    /// Not all target values are attainable because consensus code uses the compact format to
+    /// represent targets (see [`CompactTarget`]).
+    pub fn max_attainable(network: impl Into<Network>) -> Target {
```

Comment:
I think this should be `AsRef<Params>` instead. Though it's a bit weird to have a method that just copies a value. So maybe just drop this?

Reply:
I used `&Params` because when I tried `impl AsRef<Params>` I get
```
   --> bitcoin/src/blockdata/block.rs:578:51
    |
578 |         assert_eq!(real_decode.header.difficulty(&params), 2456598);
    |                                       ----------  ^^^^^^ the trait `AsRef<params::Params>` is not implemented for `params::Params`
    |                                       |
    |                                       required by a bound introduced by this call
```
Am I being brain dead?

Reply:
Oh, we need to impl `AsRef<Params> for Params`.

Reply:
Added impl and use `AsRef<Params>` for parameter type.